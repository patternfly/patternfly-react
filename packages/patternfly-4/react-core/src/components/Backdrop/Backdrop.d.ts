import { FunctionComponent, HTMLProps, ReactNode } from 'react';

export interface BackdropProps extends HTMLProps<HTMLDivElement> {
  children?: ReactNode;
  className?: string;
}

declare const Backdrop: FunctionComponent<BackdropProps>;

export default Backdrop;
