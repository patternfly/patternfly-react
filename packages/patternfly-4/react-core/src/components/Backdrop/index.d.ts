export { default as Backdrop, BackdropProps } from './Backdrop';
