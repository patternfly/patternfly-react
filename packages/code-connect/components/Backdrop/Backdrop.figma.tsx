import figma from '@figma/code-connect';
import { Backdrop } from '@patternfly/react-core';

// Documentation for Backdrop can be found at https://www.patternfly.org/components/backdrop

figma.connect(
  Backdrop,
  'https://www.figma.com/design/aEBBvq0J3EPXxHvv6WgDx9/PatternFly-6%3A-Components-Test?node-id=2873-2900',
  {
    example: () => <Backdrop />
  }
);
