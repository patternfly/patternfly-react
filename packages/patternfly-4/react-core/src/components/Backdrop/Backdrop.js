import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Backdrop/backdrop';

const propTypes = {
  /** content rendered inside the backdrop */
  children: PropTypes.node,
  /** additional classes added to the button */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const Backdrop = ({ className, children, ...props }) => (
  <div {...props} className={css(styles.backdrop, className)}>
    {children}
  </div>
);

Backdrop.propTypes = propTypes;
Backdrop.defaultProps = defaultProps;

export default Backdrop;
