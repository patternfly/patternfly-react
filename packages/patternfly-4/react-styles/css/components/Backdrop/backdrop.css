/* stylelint-enable */
/* stylelint-disable */
/* stylelint-enable */
.pf-c-backdrop {
  --pf-c-backdrop--ZIndex: var(--pf-global--ZIndex--2xl);
  --pf-c-backdrop--Color: var(--pf-global--BackgroundColor--dark-transparent-100);
  --pf-c-backdrop--BackdropFilter: blur(10px);
  position: fixed;
  top: 0;
  left: 0;
  z-index: var(--pf-c-backdrop--ZIndex);
  width: 100%;
  height: 100%;
  background-color: var(--pf-c-backdrop--Color);
  /* stylelint-disable-next-line */
  -webkit-backdrop-filter: var(--pf-c-backdrop--BackdropFilter);
  backdrop-filter: var(--pf-c-backdrop--BackdropFilter); }

.pf-c-backdrop__open {
  overflow: hidden; }
