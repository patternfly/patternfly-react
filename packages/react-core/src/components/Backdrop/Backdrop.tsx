import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Backdrop/backdrop';

export interface BackdropProps extends React.HTMLProps<HTMLDivElement> {
  /** content rendered inside the backdrop */
  children?: React.ReactNode;
  /** additional classes added to the button */
  className?: string;
}

export const Backdrop: React.FunctionComponent<BackdropProps> = ({
  children = null,
  className = '',
  ...props
}: BackdropProps) => (
  <div {...props} className={css(styles.backdrop, className)}>
    {children}
  </div>
);
Backdrop.displayName = 'Backdrop';
