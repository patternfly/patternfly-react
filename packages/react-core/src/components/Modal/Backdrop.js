import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly-next/components/Backdrop/styles.css';

const propTypes = {
  /** content rendered inside the bullseye */
  children: PropTypes.node,
  /** additional classes added to the button */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const Backdrop = ({ className, children, ...props }) => (
  <div {...props} className={css(styles.backdrop, className)}>
    {children}
  </div>
);

Backdrop.propTypes = propTypes;
Backdrop.defaultProps = defaultProps;

export default Backdrop;
