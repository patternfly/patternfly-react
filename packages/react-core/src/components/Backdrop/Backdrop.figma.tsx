import React from 'react';
import { Backdrop } from './Backdrop';
import figma from '@figma/code-connect';

/**
 * -- This file was auto-generated by Code Connect --
 * `props` includes a mapping from Figma properties and variants to
 * suggested values. You should update this to match the props of your
 * code component, and update the `example` function to return the
 * code example you'd like to see in Figma
 */

figma.connect(
  Backdrop,
  'https://www.figma.com/design/VMEX8Xg2nzhBX8rfBx53jp/branch/H3LonYnwH26v9zNEa2SXFk/PatternFly-6%3A-Components?m=auto&node-id=2873-2900',
  {
    props: {},
    example: () => <Backdrop />
  }
);
