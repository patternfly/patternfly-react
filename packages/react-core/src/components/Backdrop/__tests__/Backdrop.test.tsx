import React from 'react';
import { shallow } from 'enzyme';
import { Backdrop } from '../Backdrop';

test('Backdrop Test', () => {
  const view = shallow(<Backdrop>Backdrop</Backdrop>);
  expect(view).toMatchSnapshot();
});
