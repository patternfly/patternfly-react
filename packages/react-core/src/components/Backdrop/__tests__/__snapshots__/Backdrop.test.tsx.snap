// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Test 1`] = `
<div
  className="pf-c-backdrop"
>
  Backdrop
</div>
`;
